rhine/src/FRP/Rhine/ResamplingBuffer/FIFO.hs

```diff
@@ -23,6 +23,17 @@ fifo = timelessResamplingBuffer AsyncMealy {..} empty
       EmptyR   -> return (Nothing, empty)
       as' :> a -> return (Just a , as'  )
 
+-- |  A bounded FIFO buffer that forgets the oldest values when the size is above a given threshold.
+--   If the buffer is empty, it will return 'Nothing'.
+boundedFifo :: Monad m => Int -> ResamplingBuffer m cl1 cl2 a (Maybe a)
+boundedFifo threshold = timelessResamplingBuffer AsyncMealy {..} empty
+  where
+    amPut as a
+      | (length as) >= threshold = return $ a <| (take threshold as)
```

Comment:
* The order of `take` and left-appending `a` should be swapped, otherwise the buffer will be `threshold + 1` elements long.
* You don't need to check `length as` since `take` won't modify the sequence if it's smaller than `threshold`.

Reply:
Cool, thank you! I will update this and the whitespace issues now.

Reply:
Just fyi if you're interested: Take a look at http://hackage.haskell.org/package/containers-0.5.11.0/docs/src/Data.Sequence.Internal.html#take, it already checks the length. (Very often, source code in standard modules is very readable ;) )

Reply:
That's really useful, thank you, I appreciate any help or tips!

Reply:
I've had a go at updating as per your recommendations, is that what you were looking for?